### DISCUSSION TITLE
When's the next release?

### DISCUSSION THREAD

**Discussion by Aayush-N** [pyodide/pyodide#1300] [General]
Hello,

I've run into some difficulties in catching errors with runPythonAsync and on searching, I found the use of the loadPackagesFromImports API + runPython a better alternative,  but loadPackagesFromImports isn't available in the current release, just wondering when do you guys plan to have the next release?

Thanks!

**Comment by rth** [comment]
Likely within the next month. We need to finalize the type conversions refactoring/design improvements https://github.com/iodide-project/pyodide/pull/1233 and ideally migrate to a separate org https://github.com/iodide-project/pyodide/issues/1217

**Reply by Aayush-N** [reply]
Oh, okay, thanks a bunch for letting me know! 
Just a follow-up question, for me, the current runPythonAsync just throws an error in the console and stops execution, it doesn't reject the promise so that catch gets executed. Here is my code, is this correct? I'd be grateful if you could let me know how to handle errors in execution if not this way.

Here is my code:
`
pyodide.runPythonAsync(code.value).then(output => showCode(output)).catch((err) => {showError(err));
`

**Reply by hoodmane** [reply]
In the released versions there isn't really a way to handle errors. When `runPython` exits with an uncaught Python error, no Javascript error is thrown all that happens is a message is logged to the console. As a workaround, you could try something like:

```js
function throwError(e){
   throw new Error(e);
}
pyodide.runPython(`
import pyodide
import traceback
from js import throwError
def my_runner(code):
   try:
       pyodide.eval_code(code)
   except BaseException as e:
       s = traceback.format_exception(type(e), e, e.__traceback__)
       throwError(s)
`);
pyodide.globals.my_runner(your_code)
```
